src/Message.ts

```diff
@@ -42,45 +47,43 @@ export default class Message implements proto.MessageV1 {
   id: string
   private bytes: Uint8Array
 
+  constructor(id: string, bytes: Uint8Array, obj: proto.Message) {
+    ;[this.headerBytes, this.ciphertext] = headerBytesAndCiphertext(obj)
+    this.id = id
+    this.bytes = bytes
+  }
+
+  toBytes(): Uint8Array {
+    return this.bytes
+  }
+}
+
+// Message header carries the sender and recipient keys used to protect message.
+// Message timestamp is set by the sender.
+export class MessageV1 extends Message implements proto.MessageV1 {
+  header: proto.MessageHeaderV1 // eslint-disable-line camelcase
+
   constructor(
     id: string,
     bytes: Uint8Array,
     obj: proto.Message,
     header: proto.MessageHeaderV1
   ) {
-    const msg = extractV1Message(obj)
-    this.id = id
-    this.bytes = bytes
-    this.headerBytes = msg.headerBytes
+    super(id, bytes, obj)
     this.header = header
-    if (!msg.ciphertext) {
-      throw new Error('missing message ciphertext')
-    }
-    this.ciphertext = new Ciphertext(msg.ciphertext)
-  }
-
-  toBytes(): Uint8Array {
-    return this.bytes
   }
 
   static async create(
     obj: proto.Message,
     header: proto.MessageHeaderV1,
     bytes: Uint8Array
-  ): Promise<Message> {
+  ): Promise<MessageV1> {
     const id = bytesToHex(await sha256(bytes))
-    return new Message(id, bytes, obj, header)
-  }
-
-  static async fromBytes(bytes: Uint8Array): Promise<Message> {
-    const msg = proto.Message.decode(bytes)
-    const innerMessage = extractV1Message(msg)
-    const header = proto.MessageHeaderV1.decode(innerMessage.headerBytes)
-    return Message.create(msg, header, bytes)
+    return new MessageV1(id, bytes, obj, header)
   }
 
-  get sent(): Date | undefined {
-    return this.header ? new Date(this.header?.timestamp.toNumber()) : undefined
+  get sent(): Date {
```

Comment:
Definitely a positive change here. That `Date | undefined` was always annoying as an application developer.